### PR TITLE
Build the wrapper error payload with rapidjson

### PR DESCRIPTION
**Pull request by Archdoog**
## Summary

- The wrapper built its error payload by string concatenation, with the message
  interpolated raw. `error_json` now builds it with
  `rapidjson::writer_wrapper_t`, which is valhalla's own helper in
  `valhalla/baldr/rapidjson_utils.h`.
- The message is not always ours. Valhalla copies caller-supplied text into it
  verbatim. An unrecognized action name comes back inside error 144 that
  carries that name (`src/valhalla/src/worker.cc:1192`, then
  `src/valhalla/src/exceptions.cc:172` appends it). A request with a quote, a
  backslash, or a control character made a document the platform layer cannot
  parse. The decoder then failed, and it hid the real error. A newline also
  forged a line in the next `printf`.
- Both platform branches kept their own copy of the ladder. Both now call the
  one helper. The writer builds the object as well as the string, so no catch
  site can produce malformed JSON by hand.
- Also adds the missing newline to the unknown-exception `printf`.

### Notes for a reviewer

- **The build needs no change.** `valhalla/worker.h` already reaches
  `valhalla/baldr/rapidjson_fwd.h`, which includes `<rapidjson/fwd.h>`. The
  rapidjson header path is therefore already on this file. rapidjson is also a
  direct vcpkg dependency in `src/vcpkg.json`.
- **Why this helper, and not the other two.** `valhalla::serialize_error`
  (`valhalla/worker.h:80`) writes a different shape, with `status`,
  `status_code`, `error`, and `error_code`. It would break the `ValhallaError`
  decoder on both clients. It also wants an `Api&` that the catch site does not
  hold. It records errors and stats on that `Api`, and it covers only
  `valhalla_exception_t`. `valhalla::baldr::json` escapes as well, but it
  passes a plain `char` to `iscntrl`. The standard leaves that undefined for a
  byte above `0x7f` on a signed-char target, and the `x86` and `x86_64` ABIs
  are signed-char targets. Its escaper also carries a
  `// TODO: this may need to get more complicated`.
- **`code` is `int64_t`** because `valhalla_exception_t::code` is `unsigned`
  and the two generic cases use -1.
- **Invalid UTF-8 still passes through unescaped.** rapidjson does not validate
  encoding unless you pass `kWriteValidateEncodingFlag`. This change is about
  escaping, not about encoding, so it neither fixes nor worsens that case.

### Relation to #92

#92 fixes the same escaping bug with a hand-written escape ladder. This is an
alternative for that one bug, and the two cannot both land.

#92 also carries two fixes this PR does not: the invalid UTF-8 crash in
`ValhallaWrapper.mm`, and the request copy in `ValhallaActor::route`. Both are
independent of the escaping question and worth taking either way.

## Manual Testing

CI covers this change on both clients. The iOS `test` job sets
`VALHALLA_MOBILE_DEV: "true"` and takes the freshly built xcframework, and the
Android `connected-check` job runs `connectedCheck` on an emulator with the
fresh jniLibs. A plain local `swift test` does not, because `Package.swift:5`
falls back to the published `0.5.1` binary. Build first if you run it by hand.

**Happy path**

1. Build the wrapper: `./build.sh`.
2. Run the Apple suite against that build:
   `VALHALLA_MOBILE_DEV=true xcodebuild test -scheme ValhallaMobile -destination 'platform=iOS Simulator,name=iPhone 16'`.
3. Run the Android suite against that build:
   `cd android && ./gradlew connectedCheck`.
4. Expect the new `testErrorMessageEscapesCallerText` to pass on both, next to
   the existing tests.

**Key behaviors**

- The new test sends a costing name that holds a quote, a backslash, and a
  newline. Valhalla quotes that name back in error 125. The test parses the
  payload, then checks the code and the message. Before this change the parse
  failed, so the test captures the bug.
- `testNoSuitableEdges` still asserts `.valhallaError(171, "No suitable edges
  near location")`. The normal error path is unchanged.
- Read the log line for a thrown route. The `printf` output does not change.

**Edge cases**

- Repeat the new test with a control character such as `0x1b` in the costing
  name.
- Send a costing name with valid multi-byte UTF-8, for example an accent. The
  bytes must survive unchanged.

## Verification

No `just pre-pr` recipe exists in this repo, so the standard gate did not run.
I tried to file the tracking issue the workflow asks for, but the sandbox
blocked issue creation. Please file it, or tell me to retry with permission.

What I did check:

- **Syntax.** `clang++ -std=c++20 -fsyntax-only` on the real translation unit,
  against the header set shipped in the `valhalla-wrapper.xcframework`. The
  Apple branch is clean. The same command on `main` is clean, as a control.
- **Output equality.** A harness compiled against the vendored rapidjson
  compared the old payload with the new one. It covered all 256 byte values.
  It also covered quote injection, a trailing backslash, an embedded newline,
  an embedded NUL, and a lone continuation byte. It covered a truncated
  three-byte sequence, `0x7f`, valid multi-byte UTF-8, `code` at `UINT_MAX`,
  and `code` at -1. Every payload parses. The message round-trips byte for byte. The only
  difference is the case of the hex digits in a `\u` escape, which rapidjson
  writes in upper case. Both decode to the same value.
- **The helper swap changes no output.** A second harness compared
  `writer_wrapper_t` with the hand-held `Writer` over the same corpus. The two
  agree byte for byte, the embedded NUL case included.
- **Not checked: the Android branch.** I have no NDK here, and faking
  `__ANDROID__` against the macOS libc++ fails inside libc++ before it reaches
  this code. The catch bodies are the same text as the Apple ones. CI covers
  the four Android ABIs.

This PR closes no issue. It is an alternative implementation, not a tracked
work item.

### CODE REVIEW

File: src/wrapper/main.cpp [hunk 1]
```diff
@@ -3,6 +3,42 @@
...
+// verbatim. An unrecognized action name becomes error 144 that carries that
+// name (src/valhalla/src/worker.cc:1192, src/valhalla/src/exceptions.cc:172).
+// A quote or a control character in the message makes a document the platform
+// layer cannot parse, so the real error is lost behind a decode failure. The
+// writer escapes the message, and it also builds the object, so no caller can
+// write malformed JSON by hand.
+//
+// The code parameter is int64_t because valhalla_exception_t::code is
+// unsigned, and the other two cases use -1.
+static std::string error_json(int64_t code, const std::string& message) {
```
Comment (Archdoog) [inline src/wrapper/main.cpp]: Is there already a valhalla helper for this that we can make use of? Understood if it's internal only. 

Or an error type/struct that's serializable.

**Review by HarelM — APPROVED** [review]

**Comment by HarelM** [comment]
LGTM!
